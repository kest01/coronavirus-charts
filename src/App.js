import React from 'react';
import {connect} from 'react-redux';
import * as actions from "./redux/actions";
import Tabs from './components/tabs'
import { Preloader } from './components/preloader'
import './App.css';

import type { AppStore } from './redux/reducers'
import type { Actions } from './redux/actions'

class App extends React.Component<Actions & { store: AppStore}> {

    componentDidMount(): void {
        this.props.loadDataAction()
    }

    render() {
        if (!this.props.store.initialized) {
            return <Preloader/>
        } else {
            return (
                <div className="App">
                    <Tabs {...this.props}/>
                </div>
            );
        }
    }
}

const mapStateToProps = state => ({
    ...state
});

const mapDispatchToProps = {
    ...actions
};

const mergeProps = (stateProps, dispatchProps) => {
    return {
        ...stateProps,
        ...dispatchProps,
    }
};

export default connect(mapStateToProps, mapDispatchToProps, mergeProps)(App);
