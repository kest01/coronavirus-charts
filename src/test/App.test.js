import React from 'react';
import App from '../App';
import { createMockStoreFactory, mountConnectedComponent, snapshotify } from './utils/test-utils'

describe('Render and test App component',()=>{
  const initialState = {output:10};
  const mockStoreFactory = createMockStoreFactory();
  let store, wrapper;

  beforeEach(()=>{
    store = mockStoreFactory(initialState);
    wrapper = mountConnectedComponent(App, store);
  });

  it('Render App in loading state (snapshot)', () => {
    expect(snapshotify(wrapper)).toMatchSnapshot();
  });

});
