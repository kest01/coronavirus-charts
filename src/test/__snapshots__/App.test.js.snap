// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render and test App component Render App in loading state (snapshot) 1`] = `"<div id=\\"cube-loader\\"><div class=\\"caption\\"><div class=\\"cube-loader\\"><div class=\\"cube loader-1\\"></div><div class=\\"cube loader-2\\"></div><div class=\\"cube loader-4\\"></div><div class=\\"cube loader-3\\"></div></div></div></div>"`;

exports[`Render and test App component Render App with data (first two tabs) 1`] = `"<div class=\\"App\\"><div class=\\"react-tabs\\" data-tabs=\\"true\\"><ul class=\\"react-tabs__tab-list\\" role=\\"tablist\\"><li class=\\"react-tabs__tab react-tabs__tab--selected\\" role=\\"tab\\" id=\\"react-tabs-0\\" aria-selected=\\"true\\" aria-disabled=\\"false\\" aria-controls=\\"react-tabs-1\\" tabindex=\\"0\\">Global view</li><li class=\\"react-tabs__tab\\" role=\\"tab\\" id=\\"react-tabs-2\\" aria-selected=\\"false\\" aria-disabled=\\"false\\" aria-controls=\\"react-tabs-3\\">Favorite countries</li><li class=\\"react-tabs__tab\\" role=\\"tab\\" id=\\"react-tabs-4\\" aria-selected=\\"false\\" aria-disabled=\\"false\\" aria-controls=\\"react-tabs-5\\">Details</li><li class=\\"react-tabs__tab\\" role=\\"tab\\" id=\\"react-tabs-6\\" aria-selected=\\"false\\" aria-disabled=\\"false\\" aria-controls=\\"react-tabs-7\\">Comparison</li></ul><div class=\\"react-tabs__tab-panel react-tabs__tab-panel--selected\\" role=\\"tabpanel\\" id=\\"react-tabs-1\\" aria-labelledby=\\"react-tabs-0\\"><table class=\\"ReactTable\\" role=\\"table\\"><thead><tr role=\\"row\\"><th colspan=\\"1\\" role=\\"columnheader\\" title=\\"Toggle SortBy\\" style=\\"cursor: pointer;\\">Country</th><th colspan=\\"1\\" role=\\"columnheader\\" title=\\"Toggle SortBy\\" style=\\"cursor: pointer;\\">Total cases<span class=\\"sort-asc\\"></span></th><th colspan=\\"1\\" role=\\"columnheader\\" title=\\"Toggle SortBy\\" style=\\"cursor: pointer;\\">New cases</th><th colspan=\\"1\\" role=\\"columnheader\\" title=\\"Toggle SortBy\\" style=\\"cursor: pointer;\\">Total deaths</th><th colspan=\\"1\\" role=\\"columnheader\\" title=\\"Toggle SortBy\\" style=\\"cursor: pointer;\\">Last deaths</th><th colspan=\\"1\\" role=\\"columnheader\\" title=\\"Toggle SortBy\\" style=\\"cursor: pointer;\\">Total recovered</th><th colspan=\\"1\\" role=\\"columnheader\\" title=\\"Toggle SortBy\\" style=\\"cursor: pointer;\\">Last recovered</th><th colspan=\\"1\\" role=\\"columnheader\\" title=\\"Toggle SortBy\\" style=\\"cursor: pointer;\\">Total active</th><th colspan=\\"1\\" role=\\"columnheader\\" title=\\"Toggle SortBy\\" style=\\"cursor: pointer;\\">Active change</th></tr></thead><tbody role=\\"rowgroup\\"><tr role=\\"row\\"><td role=\\"cell\\"><button class=\\"open-country-button\\">US</button><button title=\\"Add to comparison\\" class=\\"add-button\\">+</button></td><td class=\\"global-table-column global-table-column-1\\" role=\\"cell\\">461,437</td><td class=\\"global-table-column global-table-column-2\\" role=\\"cell\\">+32,385</td><td class=\\"global-table-column global-table-column-3\\" role=\\"cell\\">16,478</td><td class=\\"global-table-column global-table-column-4\\" role=\\"cell\\">+1,783</td><td class=\\"global-table-column global-table-column-5\\" role=\\"cell\\">25,410</td><td class=\\"global-table-column global-table-column-6\\" role=\\"cell\\">+1,851</td><td class=\\"global-table-column global-table-column-7\\" role=\\"cell\\">419,549</td><td class=\\"global-table-column global-table-column-8\\" role=\\"cell\\">+28,751</td></tr><tr role=\\"row\\"><td role=\\"cell\\"><button class=\\"open-country-button\\">Italy</button><button title=\\"Add to comparison\\" class=\\"add-button\\">+</button></td><td class=\\"global-table-column global-table-column-1\\" role=\\"cell\\">143,626</td><td class=\\"global-table-column global-table-column-2\\" role=\\"cell\\">+4,204</td><td class=\\"global-table-column global-table-column-3\\" role=\\"cell\\">18,279</td><td class=\\"global-table-column global-table-column-4\\" role=\\"cell\\">+610</td><td class=\\"global-table-column global-table-column-5\\" role=\\"cell\\">28,470</td><td class=\\"global-table-column global-table-column-6\\" role=\\"cell\\">+1,979</td><td class=\\"global-table-column global-table-column-7\\" role=\\"cell\\">96,877</td><td class=\\"global-table-column global-table-column-8\\" role=\\"cell\\">+1,615</td></tr><tr role=\\"row\\"><td role=\\"cell\\"><button class=\\"open-country-button\\">France</button><button title=\\"Add to comparison\\" class=\\"add-button\\">+</button></td><td class=\\"global-table-column global-table-column-1\\" role=\\"cell\\">118,781</td><td class=\\"global-table-column global-table-column-2\\" role=\\"cell\\">+4,822</td><td class=\\"global-table-column global-table-column-3\\" role=\\"cell\\">12,228</td><td class=\\"global-table-column global-table-column-4\\" role=\\"cell\\">+1,341</td><td class=\\"global-table-column global-table-column-5\\" role=\\"cell\\">23,413</td><td class=\\"global-table-column global-table-column-6\\" role=\\"cell\\">+1,961</td><td class=\\"global-table-column global-table-column-7\\" role=\\"cell\\">83,140</td><td class=\\"global-table-column global-table-column-8\\" role=\\"cell\\">+1,520</td></tr><tr role=\\"row\\"><td role=\\"cell\\"><button class=\\"open-country-button\\">Germany</button><button title=\\"Add to comparison\\" class=\\"add-button\\">+</button></td><td class=\\"global-table-column global-table-column-1\\" role=\\"cell\\">118,181</td><td class=\\"global-table-column global-table-column-2\\" role=\\"cell\\">+4,885</td><td class=\\"global-table-column global-table-column-3\\" role=\\"cell\\">2,607</td><td class=\\"global-table-column global-table-column-4\\" role=\\"cell\\">+258</td><td class=\\"global-table-column global-table-column-5\\" role=\\"cell\\">52,407</td><td class=\\"global-table-column global-table-column-6\\" role=\\"cell\\">+6,107</td><td class=\\"global-table-column global-table-column-7\\" role=\\"cell\\">63,167</td><td class=\\"global-table-column global-table-column-8\\" role=\\"cell\\">-1,480</td></tr><tr role=\\"row\\"><td role=\\"cell\\"><button class=\\"open-country-button\\">China</button><button title=\\"Add to comparison\\" class=\\"add-button\\">+</button></td><td class=\\"global-table-column global-table-column-1\\" role=\\"cell\\">82,883</td><td class=\\"global-table-column global-table-column-2\\" role=\\"cell\\">+74</td><td class=\\"global-table-column global-table-column-3\\" role=\\"cell\\">3,339</td><td class=\\"global-table-column global-table-column-4\\" role=\\"cell\\">+2</td><td class=\\"global-table-column global-table-column-5\\" role=\\"cell\\">77,679</td><td class=\\"global-table-column global-table-column-6\\" role=\\"cell\\">+112</td><td class=\\"global-table-column global-table-column-7\\" role=\\"cell\\">1,865</td><td class=\\"global-table-column global-table-column-8\\" role=\\"cell\\">-40</td></tr><tr role=\\"row\\"><td role=\\"cell\\"><button class=\\"open-country-button\\">Iran</button><button title=\\"Add to comparison\\" class=\\"add-button\\">+</button></td><td class=\\"global-table-column global-table-column-1\\" role=\\"cell\\">66,220</td><td class=\\"global-table-column global-table-column-2\\" role=\\"cell\\">+1,634</td><td class=\\"global-table-column global-table-column-3\\" role=\\"cell\\">4,110</td><td class=\\"global-table-column global-table-column-4\\" role=\\"cell\\">+117</td><td class=\\"global-table-column global-table-column-5\\" role=\\"cell\\">32,309</td><td class=\\"global-table-column global-table-column-6\\" role=\\"cell\\">+2,497</td><td class=\\"global-table-column global-table-column-7\\" role=\\"cell\\">29,801</td><td class=\\"global-table-column global-table-column-8\\" role=\\"cell\\">-980</td></tr><tr role=\\"row\\"><td role=\\"cell\\"><button class=\\"open-country-button\\">United Kingdom</button><button title=\\"Add to comparison\\" class=\\"add-button\\">+</button></td><td class=\\"global-table-column global-table-column-1\\" role=\\"cell\\">65,872</td><td class=\\"global-table-column global-table-column-2\\" role=\\"cell\\">+4,398</td><td class=\\"global-table-column global-table-column-3\\" role=\\"cell\\">7,993</td><td class=\\"global-table-column global-table-column-4\\" role=\\"cell\\">+882</td><td class=\\"global-table-column global-table-column-5\\" role=\\"cell\\">359</td><td class=\\"global-table-column global-table-column-6\\" role=\\"cell\\">+14</td><td class=\\"global-table-column global-table-column-7\\" role=\\"cell\\">57,520</td><td class=\\"global-table-column global-table-column-8\\" role=\\"cell\\">+3,502</td></tr><tr role=\\"row\\"><td role=\\"cell\\"><button class=\\"open-country-button\\">Austria</button><button title=\\"Add to comparison\\" class=\\"add-button\\">+</button></td><td class=\\"global-table-column global-table-column-1\\" role=\\"cell\\">13,244</td><td class=\\"global-table-column global-table-column-2\\" role=\\"cell\\">+302</td><td class=\\"global-table-column global-table-column-3\\" role=\\"cell\\">295</td><td class=\\"global-table-column global-table-column-4\\" role=\\"cell\\">+22</td><td class=\\"global-table-column global-table-column-5\\" role=\\"cell\\">5,240</td><td class=\\"global-table-column global-table-column-6\\" role=\\"cell\\">+728</td><td class=\\"global-table-column global-table-column-7\\" role=\\"cell\\">7,709</td><td class=\\"global-table-column global-table-column-8\\" role=\\"cell\\">-448</td></tr><tr role=\\"row\\"><td role=\\"cell\\"><button class=\\"open-country-button\\">Russia</button><button title=\\"Add to comparison\\" class=\\"add-button\\">+</button></td><td class=\\"global-table-column global-table-column-1\\" role=\\"cell\\">10,131</td><td class=\\"global-table-column global-table-column-2\\" role=\\"cell\\">+1,459</td><td class=\\"global-table-column global-table-column-3\\" role=\\"cell\\">76</td><td class=\\"global-table-column global-table-column-4\\" role=\\"cell\\">+13</td><td class=\\"global-table-column global-table-column-5\\" role=\\"cell\\">698</td><td class=\\"global-table-column global-table-column-6\\" role=\\"cell\\">+118</td><td class=\\"global-table-column global-table-column-7\\" role=\\"cell\\">9,357</td><td class=\\"global-table-column global-table-column-8\\" role=\\"cell\\">+1,328</td></tr><tr role=\\"row\\"><td role=\\"cell\\"><button class=\\"open-country-button\\">Israel</button><button title=\\"Add to comparison\\" class=\\"add-button\\">+</button></td><td class=\\"global-table-column global-table-column-1\\" role=\\"cell\\">9,968</td><td class=\\"global-table-column global-table-column-2\\" role=\\"cell\\">+564</td><td class=\\"global-table-column global-table-column-3\\" role=\\"cell\\">86</td><td class=\\"global-table-column global-table-column-4\\" role=\\"cell\\">+13</td><td class=\\"global-table-column global-table-column-5\\" role=\\"cell\\">1,011</td><td class=\\"global-table-column global-table-column-6\\" role=\\"cell\\">+210</td><td class=\\"global-table-column global-table-column-7\\" role=\\"cell\\">8,871</td><td class=\\"global-table-column global-table-column-8\\" role=\\"cell\\">+341</td></tr><tr role=\\"row\\"><td role=\\"cell\\"><button class=\\"open-country-button\\">Ukraine</button><button title=\\"Add to comparison\\" class=\\"add-button\\">+</button></td><td class=\\"global-table-column global-table-column-1\\" role=\\"cell\\">1,892</td><td class=\\"global-table-column global-table-column-2\\" role=\\"cell\\">+224</td><td class=\\"global-table-column global-table-column-3\\" role=\\"cell\\">57</td><td class=\\"global-table-column global-table-column-4\\" role=\\"cell\\">+5</td><td class=\\"global-table-column global-table-column-5\\" role=\\"cell\\">45</td><td class=\\"global-table-column global-table-column-6\\" role=\\"cell\\">+10</td><td class=\\"global-table-column global-table-column-7\\" role=\\"cell\\">1,790</td><td class=\\"global-table-column global-table-column-8\\" role=\\"cell\\">+209</td></tr><tr role=\\"row\\"><td role=\\"cell\\"><button class=\\"open-country-button\\">Belarus</button><button title=\\"Add to comparison\\" class=\\"add-button\\">+</button></td><td class=\\"global-table-column global-table-column-1\\" role=\\"cell\\">1,486</td><td class=\\"global-table-column global-table-column-2\\" role=\\"cell\\">+420</td><td class=\\"global-table-column global-table-column-3\\" role=\\"cell\\">16</td><td class=\\"global-table-column global-table-column-4\\" role=\\"cell\\">+3</td><td class=\\"global-table-column global-table-column-5\\" role=\\"cell\\">139</td><td class=\\"global-table-column global-table-column-6\\" role=\\"cell\\">+62</td><td class=\\"global-table-column global-table-column-7\\" role=\\"cell\\">1,331</td><td class=\\"global-table-column global-table-column-8\\" role=\\"cell\\">+355</td></tr><tr role=\\"row\\"><td role=\\"cell\\"><button class=\\"open-country-button\\">Armenia</button><button title=\\"Add to comparison\\" class=\\"add-button\\">+</button></td><td class=\\"global-table-column global-table-column-1\\" role=\\"cell\\">921</td><td class=\\"global-table-column global-table-column-2\\" role=\\"cell\\">+40</td><td class=\\"global-table-column global-table-column-3\\" role=\\"cell\\">10</td><td class=\\"global-table-column global-table-column-4\\" role=\\"cell\\">+1</td><td class=\\"global-table-column global-table-column-5\\" role=\\"cell\\">138</td><td class=\\"global-table-column global-table-column-6\\" role=\\"cell\\">+24</td><td class=\\"global-table-column global-table-column-7\\" role=\\"cell\\">773</td><td class=\\"global-table-column global-table-column-8\\" role=\\"cell\\">+15</td></tr></tbody></table><br></div><div class=\\"react-tabs__tab-panel\\" role=\\"tabpanel\\" id=\\"react-tabs-3\\" aria-labelledby=\\"react-tabs-2\\"></div><div class=\\"react-tabs__tab-panel\\" role=\\"tabpanel\\" id=\\"react-tabs-5\\" aria-labelledby=\\"react-tabs-4\\"></div><div class=\\"react-tabs__tab-panel\\" role=\\"tabpanel\\" id=\\"react-tabs-7\\" aria-labelledby=\\"react-tabs-6\\"></div></div></div>"`;

exports[`Render and test App component Render App with data (first two tabs) 2`] = `"<div class=\\"App\\"><div class=\\"react-tabs\\" data-tabs=\\"true\\"><ul class=\\"react-tabs__tab-list\\" role=\\"tablist\\"><li class=\\"react-tabs__tab\\" role=\\"tab\\" id=\\"react-tabs-0\\" aria-selected=\\"false\\" aria-disabled=\\"false\\" aria-controls=\\"react-tabs-1\\">Global view</li><li class=\\"react-tabs__tab react-tabs__tab--selected\\" role=\\"tab\\" id=\\"react-tabs-2\\" aria-selected=\\"true\\" aria-disabled=\\"false\\" aria-controls=\\"react-tabs-3\\" tabindex=\\"0\\">Favorite countries</li><li class=\\"react-tabs__tab\\" role=\\"tab\\" id=\\"react-tabs-4\\" aria-selected=\\"false\\" aria-disabled=\\"false\\" aria-controls=\\"react-tabs-5\\">Details</li><li class=\\"react-tabs__tab\\" role=\\"tab\\" id=\\"react-tabs-6\\" aria-selected=\\"false\\" aria-disabled=\\"false\\" aria-controls=\\"react-tabs-7\\">Comparison</li></ul><div class=\\"react-tabs__tab-panel\\" role=\\"tabpanel\\" id=\\"react-tabs-1\\" aria-labelledby=\\"react-tabs-0\\"></div><div class=\\"react-tabs__tab-panel react-tabs__tab-panel--selected\\" role=\\"tabpanel\\" id=\\"react-tabs-3\\" aria-labelledby=\\"react-tabs-2\\"><table class=\\"ReactTable\\" role=\\"table\\"><thead><tr role=\\"row\\"><th colspan=\\"1\\" role=\\"columnheader\\" title=\\"Toggle SortBy\\" style=\\"cursor: pointer;\\">Country</th><th colspan=\\"1\\" role=\\"columnheader\\" title=\\"Toggle SortBy\\" style=\\"cursor: pointer;\\">Total cases<span class=\\"sort-asc\\"></span></th><th colspan=\\"1\\" role=\\"columnheader\\" title=\\"Toggle SortBy\\" style=\\"cursor: pointer;\\">New cases</th><th colspan=\\"1\\" role=\\"columnheader\\" title=\\"Toggle SortBy\\" style=\\"cursor: pointer;\\">Total deaths</th><th colspan=\\"1\\" role=\\"columnheader\\" title=\\"Toggle SortBy\\" style=\\"cursor: pointer;\\">Last deaths</th><th colspan=\\"1\\" role=\\"columnheader\\" title=\\"Toggle SortBy\\" style=\\"cursor: pointer;\\">Total recovered</th><th colspan=\\"1\\" role=\\"columnheader\\" title=\\"Toggle SortBy\\" style=\\"cursor: pointer;\\">Last recovered</th><th colspan=\\"1\\" role=\\"columnheader\\" title=\\"Toggle SortBy\\" style=\\"cursor: pointer;\\">Total active</th><th colspan=\\"1\\" role=\\"columnheader\\" title=\\"Toggle SortBy\\" style=\\"cursor: pointer;\\">Active change</th></tr></thead><tbody role=\\"rowgroup\\"><tr role=\\"row\\"><td role=\\"cell\\"><button class=\\"open-country-button\\">US</button><button title=\\"Add to comparison\\" class=\\"add-button\\">+</button></td><td class=\\"global-table-column global-table-column-1\\" role=\\"cell\\">461,437</td><td class=\\"global-table-column global-table-column-2\\" role=\\"cell\\">+32,385</td><td class=\\"global-table-column global-table-column-3\\" role=\\"cell\\">16,478</td><td class=\\"global-table-column global-table-column-4\\" role=\\"cell\\">+1,783</td><td class=\\"global-table-column global-table-column-5\\" role=\\"cell\\">25,410</td><td class=\\"global-table-column global-table-column-6\\" role=\\"cell\\">+1,851</td><td class=\\"global-table-column global-table-column-7\\" role=\\"cell\\">419,549</td><td class=\\"global-table-column global-table-column-8\\" role=\\"cell\\">+28,751</td></tr><tr role=\\"row\\"><td role=\\"cell\\"><button class=\\"open-country-button\\">Italy</button><button title=\\"Add to comparison\\" class=\\"add-button\\">+</button></td><td class=\\"global-table-column global-table-column-1\\" role=\\"cell\\">143,626</td><td class=\\"global-table-column global-table-column-2\\" role=\\"cell\\">+4,204</td><td class=\\"global-table-column global-table-column-3\\" role=\\"cell\\">18,279</td><td class=\\"global-table-column global-table-column-4\\" role=\\"cell\\">+610</td><td class=\\"global-table-column global-table-column-5\\" role=\\"cell\\">28,470</td><td class=\\"global-table-column global-table-column-6\\" role=\\"cell\\">+1,979</td><td class=\\"global-table-column global-table-column-7\\" role=\\"cell\\">96,877</td><td class=\\"global-table-column global-table-column-8\\" role=\\"cell\\">+1,615</td></tr><tr role=\\"row\\"><td role=\\"cell\\"><button class=\\"open-country-button\\">Germany</button><button title=\\"Add to comparison\\" class=\\"add-button\\">+</button></td><td class=\\"global-table-column global-table-column-1\\" role=\\"cell\\">118,181</td><td class=\\"global-table-column global-table-column-2\\" role=\\"cell\\">+4,885</td><td class=\\"global-table-column global-table-column-3\\" role=\\"cell\\">2,607</td><td class=\\"global-table-column global-table-column-4\\" role=\\"cell\\">+258</td><td class=\\"global-table-column global-table-column-5\\" role=\\"cell\\">52,407</td><td class=\\"global-table-column global-table-column-6\\" role=\\"cell\\">+6,107</td><td class=\\"global-table-column global-table-column-7\\" role=\\"cell\\">63,167</td><td class=\\"global-table-column global-table-column-8\\" role=\\"cell\\">-1,480</td></tr><tr role=\\"row\\"><td role=\\"cell\\"><button class=\\"open-country-button\\">China</button><button title=\\"Add to comparison\\" class=\\"add-button\\">+</button></td><td class=\\"global-table-column global-table-column-1\\" role=\\"cell\\">82,883</td><td class=\\"global-table-column global-table-column-2\\" role=\\"cell\\">+74</td><td class=\\"global-table-column global-table-column-3\\" role=\\"cell\\">3,339</td><td class=\\"global-table-column global-table-column-4\\" role=\\"cell\\">+2</td><td class=\\"global-table-column global-table-column-5\\" role=\\"cell\\">77,679</td><td class=\\"global-table-column global-table-column-6\\" role=\\"cell\\">+112</td><td class=\\"global-table-column global-table-column-7\\" role=\\"cell\\">1,865</td><td class=\\"global-table-column global-table-column-8\\" role=\\"cell\\">-40</td></tr><tr role=\\"row\\"><td role=\\"cell\\"><button class=\\"open-country-button\\">Russia</button><button title=\\"Add to comparison\\" class=\\"add-button\\">+</button></td><td class=\\"global-table-column global-table-column-1\\" role=\\"cell\\">10,131</td><td class=\\"global-table-column global-table-column-2\\" role=\\"cell\\">+1,459</td><td class=\\"global-table-column global-table-column-3\\" role=\\"cell\\">76</td><td class=\\"global-table-column global-table-column-4\\" role=\\"cell\\">+13</td><td class=\\"global-table-column global-table-column-5\\" role=\\"cell\\">698</td><td class=\\"global-table-column global-table-column-6\\" role=\\"cell\\">+118</td><td class=\\"global-table-column global-table-column-7\\" role=\\"cell\\">9,357</td><td class=\\"global-table-column global-table-column-8\\" role=\\"cell\\">+1,328</td></tr><tr role=\\"row\\"><td role=\\"cell\\"><button class=\\"open-country-button\\">Israel</button><button title=\\"Add to comparison\\" class=\\"add-button\\">+</button></td><td class=\\"global-table-column global-table-column-1\\" role=\\"cell\\">9,968</td><td class=\\"global-table-column global-table-column-2\\" role=\\"cell\\">+564</td><td class=\\"global-table-column global-table-column-3\\" role=\\"cell\\">86</td><td class=\\"global-table-column global-table-column-4\\" role=\\"cell\\">+13</td><td class=\\"global-table-column global-table-column-5\\" role=\\"cell\\">1,011</td><td class=\\"global-table-column global-table-column-6\\" role=\\"cell\\">+210</td><td class=\\"global-table-column global-table-column-7\\" role=\\"cell\\">8,871</td><td class=\\"global-table-column global-table-column-8\\" role=\\"cell\\">+341</td></tr><tr role=\\"row\\"><td role=\\"cell\\"><button class=\\"open-country-button\\">Ukraine</button><button title=\\"Add to comparison\\" class=\\"add-button\\">+</button></td><td class=\\"global-table-column global-table-column-1\\" role=\\"cell\\">1,892</td><td class=\\"global-table-column global-table-column-2\\" role=\\"cell\\">+224</td><td class=\\"global-table-column global-table-column-3\\" role=\\"cell\\">57</td><td class=\\"global-table-column global-table-column-4\\" role=\\"cell\\">+5</td><td class=\\"global-table-column global-table-column-5\\" role=\\"cell\\">45</td><td class=\\"global-table-column global-table-column-6\\" role=\\"cell\\">+10</td><td class=\\"global-table-column global-table-column-7\\" role=\\"cell\\">1,790</td><td class=\\"global-table-column global-table-column-8\\" role=\\"cell\\">+209</td></tr><tr role=\\"row\\"><td role=\\"cell\\"><button class=\\"open-country-button\\">Belarus</button><button title=\\"Add to comparison\\" class=\\"add-button\\">+</button></td><td class=\\"global-table-column global-table-column-1\\" role=\\"cell\\">1,486</td><td class=\\"global-table-column global-table-column-2\\" role=\\"cell\\">+420</td><td class=\\"global-table-column global-table-column-3\\" role=\\"cell\\">16</td><td class=\\"global-table-column global-table-column-4\\" role=\\"cell\\">+3</td><td class=\\"global-table-column global-table-column-5\\" role=\\"cell\\">139</td><td class=\\"global-table-column global-table-column-6\\" role=\\"cell\\">+62</td><td class=\\"global-table-column global-table-column-7\\" role=\\"cell\\">1,331</td><td class=\\"global-table-column global-table-column-8\\" role=\\"cell\\">+355</td></tr></tbody></table><br></div><div class=\\"react-tabs__tab-panel\\" role=\\"tabpanel\\" id=\\"react-tabs-5\\" aria-labelledby=\\"react-tabs-4\\"></div><div class=\\"react-tabs__tab-panel\\" role=\\"tabpanel\\" id=\\"react-tabs-7\\" aria-labelledby=\\"react-tabs-6\\"></div></div></div>"`;
